Qutrit Rotations #1 (TRX) (#2845)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

<h3>New features since last release</h3>

* Added the `qml.TRX` qutrit operation, which applies an X rotation to a specified subspace.
  ([#2845](https://github.com/PennyLaneAI/pennylane/pull/2845))

<h3>Improvements</h3>

* Updated `pennylane/qnode.py` to support parameter-shift differentiation on qutrit devices.
  ([#2845](https://github.com/PennyLaneAI/pennylane/pull/2845))

> * Updated `pennylane/utils.py:sparse_hamiltonian` to include a `level` keyword argument to create
>   work with Hamiltonians with arbitrary levels per wire (qutrits, etc).

Reviewer: ```suggestion
* Updated `pennylane/utils.py:sparse_hamiltonian` to include a `level` keyword argument to 
  support Hamiltonians for systems with an arbitrary number of levels per wire (qutrits, etc).
```
